### PR TITLE
Add the remaining eudemo components and make it work

### PR DESCRIPTION
**Pull request by je-cook**
## Description

This is based off #1852 which in turn is based off #1851. Should be merged in order.

This adds the remaining components that were converted but never added back. 
Adds a bodge for the vvts that shouldnt be needed, 
Adds Cryostat, CryoTS RadShield and Power cycle.

The units on powercycle are wrong or there is some other bug but it now runs.

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: bluemira/builders/cryostat.py [hunk 1]
```diff
@@ -132,9 +132,17 @@ def build_xz(self, x_out: float, z_top: float) -> PhysicalComponent:
             x coordinate extremity
         z_top: float
             z coordinate extremity
+
+        Notes
+        -----
+        Only designed for an inward kind, outward kinks will fail
```
Comment (CoronelBuendia) [inline bluemira/builders/cryostat.py:138]:
```suggestion
        Only designed for an inward kink, outward kinks will fail
```

File: bluemira/builders/cryostat.py [hunk 2]
```diff
@@ -146,8 +154,8 @@ def build_xz(self, x_out: float, z_top: float) -> PhysicalComponent:
         x_outer[1:-1] += tk
 
         z_outer = np.array([z_bot, z_bot, z_mid, z_mid, z_top, z_top])
-        z_outer[:2] -= tk
-        z_outer[2:] += tk
+        z_outer[:4] -= tk
+        z_outer[4:] += tk
```
Comment (CoronelBuendia) [inline bluemira/builders/cryostat.py:158]: Can confirm: this is right, but I have a feeling we've either changed this recently, or changed it the same way somewhere else. Anyway, let's not touch it any more!

File: eudemo/config/params.json [hunk 3]
```diff
@@ -179,6 +179,30 @@
         "source": "Calculated",
         "long_name": "Plasma volume"
     },
+    "well_depth": {
+        "value": 5,
+        "unit": "m",
+        "source": "Input",
+        "long_name": "cryostat well depth (is the unit right?)"
```
Comment (CoronelBuendia) [inline eudemo/config/params.json:186]:
```suggestion
        "long_name": "cryostat well depth"
```
yes

File: eudemo/config/params.json [hunk 4]
```diff
@@ -179,6 +179,30 @@
...
+    },
+    "x_g_support": {
+        "value": 13,
+        "unit": "m",
+        "source": "Input",
+        "long_name": "TODO add me"
```
Comment (CoronelBuendia) [inline eudemo/config/params.json:192]:
```suggestion
        "long_name": "Gravity support radius. Used for TF and VV supports (assuming a supporting ring)"
```

File: eudemo/config/params.json [hunk 5]
```diff
@@ -179,6 +179,30 @@
...
+    },
+    "x_gs_kink_diff": {
+        "value": 2,
+        "unit": "m",
+        "source": "Input",
+        "long_name": "TODO add me"
```
Comment (CoronelBuendia) [inline eudemo/config/params.json:198]:
```suggestion
        "long_name": "Lower port kink radius offset from the gravity support radius."
```

File: eudemo/config/params.json [hunk 6]
```diff
@@ -179,6 +179,30 @@
...
+    },
+    "z_gs": {
+        "value": -15,
+        "unit": "m",
+        "source": "Input",
+        "long_name": "TODO improve this name: z gravity support"
```
Comment (CoronelBuendia) [inline eudemo/config/params.json:204]:
```suggestion
        "long_name": "Gravity support vertical position at the ground"
```

**Review by CoronelBuendia — CHANGES_REQUESTED** [review]

File: bluemira/builders/cryostat.py (hunk 2 above)
Comment (je-cook) [reply bluemira/builders/cryostat.py:158]: It was probably me in the conversion...at least its right now

**Review by CoronelBuendia — APPROVED** [review]

File: eudemo/eudemo/power_cycle.py [hunk 7]
```diff
@@ -172,11 +181,12 @@ def run(self, run_result):
         }
 
 
-class SteadyStatePowerCycleSolver(SolverABC):
+class SteadyStatePowerCycleSolver(CodesSolver):
```
Comment (hsaunders1904) [inline eudemo/eudemo/power_cycle.py:184]: Did I miss this when removing the `SolverABC`? Whoops!

This isn't really an external code solver, do we need to open an issue to revisit this at some point?

File: eudemo/eudemo/reactor.py [hunk 8]
```diff
@@ -241,4 +309,25 @@ def _read_json(file_path: str) -> Dict:
...
+
+    reactor.cryostat = build_cryostat(
+        params, build_config.get("Cryostat", {}), reactor.cryostat_thermal.xz_boundary()
+    )
+
+    reactor.radiation_shield = build_radiation_shield(
+        params, build_config.get("RadiationShield", {}), reactor.cryostat.xz_boundary()
+    )
+
+    # TODO fix units
```
Comment (hsaunders1904) [inline eudemo/eudemo/reactor.py:327]: Issue open for TODO?

**Review by hsaunders1904 — APPROVED** [review]
Again, I can't functionally test this because of my Dolfin issue, but the code looks good to me. A couple of minor things, but if you're happy it's working, I'm happy.

File: eudemo/eudemo/power_cycle.py (hunk 7 above)
Comment (je-cook) [reply eudemo/eudemo/power_cycle.py:184]: Yeah I'll open an issue

Comment (je-cook) [reply eudemo/eudemo/power_cycle.py:184]: #1923

**Pull request merged**